### DISCUSSION TITLE
Does exists a way to scan devDependencies on a npm-based repository?

### DISCUSSION THREAD

**Discussion by NicolasOmar** [aquasecurity/trivy#3714] [Q&A] [answered]
Hello team, first I want to thank you for such awsome tool and its Github Action that I've been integrating into my repository to improve security scans.
Now, getting into the topic, I have a PR to include Trivy scan for repo which is a `npm project` with no dependencies (only has `devDependencies`) and after some documentation research (because I was having no vulnerabilities finded on my tests) I understood that Trivy doesn't look only into production-specified dependencies.
There is any way to force scanning on this case or do you have in your plans to implement it as an option on future versions?

For context:
- [My repo](https://github.com/NicolasOmar/reactive-bulma)
- [The PR with Trivy's implementation](https://github.com/NicolasOmar/reactive-bulma/pull/27)
- [The issue I created to log the investigation progress](https://github.com/NicolasOmar/reactive-bulma/issues/26)

**Comment by knqyf263** [comment, marked as answer]
Trivy scans lock files and exclude devDependencies for `package-lock.json` and `pnpm-lock.yaml` as described in the following page. Are you using yarn?
https://aquasecurity.github.io/trivy/v0.37/docs/vulnerability/detection/language/

**Reply by knqyf263** [reply]
Oh, sorry I missed your repo. You use npm, so devDependencies should be excluded. @afdesk Would you take a look?
https://github.com/NicolasOmar/reactive-bulma/blob/main/package-lock.json

**Reply by afdesk** [reply]
@knqyf263 this `package-lock.json` contains only dev dependencies, and `go-dep-parser` skips it:
https://github.com/aquasecurity/go-dep-parser/blob/main/pkg/nodejs/npm/parse.go#L72-L74

should we change this behavior?

**Reply by knqyf263** [reply]
I was confused again... I thought @NicolasOmar wanted to skip dev dependencies. I'm sorry for confusing you guys.

Yes, there is no option to scan dev dependencies now as @afdesk said.

I'm not sure we want to scan dev dependencies. By default, they should be skipped.

@NicolasOmar If you want to request the feature, you can raise an issue. Let's see how many people want that.

**Reply by NicolasOmar** [reply]
Hello @knqyf263, I understand the situation and I appreciate the time to find this particular point.
I will make an issue during today and will refer to this discussion in order to give some context for that request.
Thank you again for your work and attention team.
